specification.md

```diff
@@ -2474,6 +2530,134 @@ authenticity of the downloaded file, for example can be the
 The exact signing and verification method is Agent specific. See
 [Code Signing](#code-signing) for recommendations.
 
+### Custom Messages
+
+#### Motivation
+
+The OpAMP protocol is intended to cover all aspects of remote management and configuration
+of Agents. If more requirements within this core functionality are identified, it is
+expected that the protocol will be extended to support this additional behavior in future
+releases. However, there are some use cases that require communication with Agents that
+will not by supported by the OpAMP protocol because they are not within the scope of Agent
+remote management. The CustomMessage allows custom behavior to be implemented between a
+Server and Agent without having to open another connection or define an entirely new
+protocol.
+
+While interoperability between many Agents and Agent Management Servers is a goal of
+OpAMP, additional custom functionality may only be supported by specific Agents and
+Servers. Where appropriate, implementers are encouraged to document their use of
+CustomMessage so that support can be added to additional platforms.
+
+#### CustomMessageCapabilities Message
+
+The Agent and Server should both use this message to signal that they support specific
```

Comment:
Just a thought. Although probably unlikely, a string capability for one server may differ from the functionality on another (e.g. vendor) and a connecting agent may not know any better resulting in unexpected behaviour. Do we care about interoperability? Do the strings need to be distro/vendor specific? Do we need another field?

Reply:
The intent is that they would use a reverse FQDN to be vendor specific, e.g. our pause/resume feature might be `com.observiq.pause`. I used `io.opentelemetry` in the examples to keep vendor specific names out of the spec.